src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java

```diff
@@ -136,101 +131,39 @@ StageWithCaseData getActiveStage(UUID caseUUID, UUID stageUUID) {
-        final AtomicBoolean isRecreateOfCurrentlyActive = new AtomicBoolean(false);
-        final AtomicReference<Stage> stageToActivate = new AtomicReference<>(null);
-
-        existingStageOfRequestedType.ifPresentOrElse((Stage stage) -> {
-            log.debug("Stage type of {} exists for caseUUID: {}, recreating stage",createStageRequest.getType(), caseUUID);
-            stageToActivate.set(stage);
-            isRecreation.set(true);
-            isRecreateOfCurrentlyActive.set(activeStages.contains(stage));
-        },() -> {
-            log.debug("Stage type of {} does not currently exist for caseUUID: {}, creating new stage",createStageRequest.getType(), caseUUID);
-            stageToActivate.set(new Stage(caseUUID, createStageRequest.getType(),
-                    null, null, createStageRequest.getTransitionNoteUUID()));
-        });
 
-        CaseData caseData = caseDataService.getCase(caseUUID);
-        calculateDeadlines(stageToActivate.get(), caseData);
-
-        stageRepository.save(stageToActivate.get());
-        log.info("Created Stage: {}, Type: {}, Case: {}", stageToActivate.get().getUuid(), stageToActivate.get().getStageType(), stageToActivate.get().getCaseUUID(), value(EVENT, STAGE_CREATED));
-
-        caseDataService.updateCaseData(caseData, stageToActivate.get().getUuid(), Map.of(CaseworkConstants.CURRENT_STAGE, stageToActivate.get().getStageType()));
-
-        List<UUID> assignedUserUUIDList = activeStages.stream().map(BaseStage::getUserUUID).collect(Collectors.toList());
-        UUID currentUserUUID = null;
-
-        if (!assignedUserUUIDList.isEmpty()) {
-            currentUserUUID = assignedUserUUIDList.get(0);
-        }
-
-        assignTeamAndMemberUserToStage(stageToActivate.get(), createStageRequest.getTeamUUID(), createStageRequest.getUserUUID());
-
-        // deactivate old active stage/s if new stage not current stage
-        if (!isRecreateOfCurrentlyActive.get()) {
-            activeStages.forEach(value -> updateStageTeam(value, caseData.getDataMap(), caseData.getReference(), null, null));
-        }
-
-        // Update audit and timeline unless active and allocated to same user and team - positioned here maintains timeline consistency.
-        if (isRecreation.get() && !isRecreateOfCurrentlyActive.get()) {
-            auditClient.recreateStage(stageToActivate.get());
-        } else if (!isRecreation.get()) {
-            auditClient.createStage(stageToActivate.get());
-        }
+        // Store the current user before clearing it later
+        UUID currentUserUUID = allStages
+                .stream()
+                .map(BaseStage::getUserUUID)
+                .findFirst()
+                .orElse(null);
 
-        updateAssignmentAudit(stageToActivate.get());
+        UUID newTeamUUID = Objects.requireNonNullElse(
+                createStageRequest.getTeamUUID(),
+                infoClient.getTeamForStageType(createStageRequest.getType()).getUuid());
```

Comment:
What if there isn't one? Null would be returned and lead to a case without a active stage. 

We need to go back and look into this further. There are workflows such as POGR and MPAM that don't work on the principle of one team owns that stage. Which is by design for security. 

Do they need a overarching default team? Thats a question for another day.

Reply:
this is literally the 'bug' we've been having, there should be a team in stage in the diagrams and a 'fallback' team in the database for lost cases. there is no fallback stage for the TO cases at least.

Reply:
This is knowledge that was lost in the passage of time. And inflexible as only supports the DCU workflow. We probably need to look at a better solution.

Reply:
The original intent was that casework should be 'dumb'; so the calls to info are not in the right place and are the concern of the workflow service. moving them into casework hasn't solved anything wrt cases falling off the workstacks

Reply:
I disagree, Casework is the powerhouse as Workflow should be just a ACL of Camunda and shouldn't know anything about our implementation and should do basic things i.e. 'move forward', clear X'